x-pack/test/apm_api_integration/tests/alerts/rule_registry.ts

```diff
@@ -104,10 +115,12 @@ export default function ApiTest({ getService }: FtrProviderContext) {
       await new Promise((resolve) => {
         setTimeout(resolve, BULK_INDEX_DELAY);
       });
-      await es.indices.refresh({
-        index: ALERTS_INDEX_TARGET,
-      });
-
+      try {
+        await es.indices.refresh({
+          index: targetIndices[0],
+        });
+        // eslint-disable-next-line no-empty
+      } catch (exc) {}
```

Comment:
can you leave a comment here why it's okay for this to fail?

Reply:
For sure I can add a comment to the code. Essentially it boils down to this:

When calling refresh on an index pattern `.alerts-observability.apm.alerts*` The response from Elasticsearch is a 200.

When calling refresh on a concrete index alias `.alerts-observability.apm.alerts-default` for instance, we receive a 404 error `index_not_found_exception`

Since we are receiving a concrete index alias from the observability api instead of a kibana index pattern and we understand / expect that this index does not exist at certain points of the test, we can try-catch at certain points without caring if the call fails. There are points in the code where we do want to ensure we get the appropriate error message back (such as this `expect` [you commented on](https://github.com/elastic/kibana/pull/109715#discussion_r694901480))

<details><summary>Example..</summary>

```
# This request yields a 200 response
GET thisdoesnotexist*/_refresh

# Response
{
  "_shards" : {
    "total" : 0,
    "successful" : 0,
    "failed" : 0
  }
}


# This request yields a 404 error

GET thisdoesnotexist/_refresh

# Response
{
  "error" : {
    "root_cause" : [
      {
        "type" : "index_not_found_exception",
        "reason" : "no such index [thisdoesnotexist]",
        "resource.type" : "index_or_alias",
        "resource.id" : "thisdoesnotexist",
        "index_uuid" : "_na_",
        "index" : "thisdoesnotexist"
      }
    ],
    "type" : "index_not_found_exception",
    "reason" : "no such index [thisdoesnotexist]",
    "resource.type" : "index_or_alias",
    "resource.id" : "thisdoesnotexist",
    "index_uuid" : "_na_",
    "index" : "thisdoesnotexist"
  },
  "status" : 404
}
```

</details>